WordPress/Classes/Utility/WPAuthTokenIssueSolver.m

```diff
@@ -0,0 +1,133 @@
+ */
+- (BOOL)hasAuthTokenIssues
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+    AccountService *accountService = [[AccountService alloc] initWithManagedObjectContext:context];
+    WPAccount *account = [accountService defaultWordPressComAccount];
+    
+    BOOL hasAuthTokenIssues = account && ![account authToken];
+    
+    return hasAuthTokenIssues;
+}
+
+#pragma mark - Alerts
+
+/**
+ *  @brief      Shows the alert when the re-authentication is cancelled by the user.
+ *
+ *  @param      okBlock     The block that will be executed if the user confirms the operation.
+ */
+- (void)showCancelReAuthenticationAlertAndOnOK:(WPAuthTokenissueSolverCompletionBlock)okBlock
+{
+    NSParameterAssert(okBlock);
+    
+    NSString *alertTitle = NSLocalizedString(@"Careful!",
+                                             @"Title for the warning shown to the user when he refuses to re-login when the authToken is missing.");
+    NSString *alertMessage = NSLocalizedString(@"If you proceed, all your WP.com account data will be removed from this device.  This means any unsaved data such as locally stored posts will be deleted.  This will not affect data that has already been uploaded to your WP.com account.",
+                                               @"Message for the warning shown to the user when he refuses to re-login when the authToken is missing.");
+    NSString *cancelButtonTitle = NSLocalizedString(@"Cancel",
+                                                    @"Cancel button title for the warning shown to the user when he refuses to re-login when the authToken is missing.");
+    NSString *deleteButtonTitle = NSLocalizedString(@"Delete",
+                                                    @"Delete button title for the warning shown to the user when he refuses to re-login when the authToken is missing.");
+    
+    [UIAlertView showWithTitle:alertTitle
+                       message:alertMessage
+             cancelButtonTitle:cancelButtonTitle
+             otherButtonTitles:@[deleteButtonTitle]
+                      tapBlock:^(UIAlertView *alertView, NSInteger buttonIndex) {
+                          if (buttonIndex == alertView.firstOtherButtonIndex) {
+                              okBlock();
+                          }
+                      }];
+}
+
+/**
+ *  @brief      Shows the alert explaining the authToken issue to the user.
+ */
+- (void)showExplanationAlertForReAuthenticationDueToMissingAuthToken
+{
+    NSString *alertTitle = NSLocalizedString(@"Oops!",
+                                             @"Title for the warning shown to the user when the app realizes there should be an auth token but there isn't one.");
+    NSString *alertMessage = NSLocalizedString(@"We have detected a synchronization problem with your locally stored WP.com credentials. You're going to be prompted to re-authenticate.",
```

Comment:
Also need to spell out WordPress.com here and probably need to reduce the technical level of the message.  Maybe something very simple like: 

> There was a problem connecting to WordPress.com.  Please sign in again.